react/features/filmstrip/components/web/Thumbnail.js

```diff
@@ -0,0 +1,639 @@
+                        tooltipPosition = { tooltipPosition } />
+                    { showDominantSpeaker && _participantCount > 2
+                        ? <DominantSpeakerIndicator
+                            iconSize = { iconSize }
+                            tooltipPosition = { tooltipPosition } />
+                        : null }
+                </AtlasKitThemeProvider>
+            </div>);
+    }
+
+    /**
+     * Renders the avatar.
+     *
+     * @returns {ReactElement}
+     */
+    _renderAvatar() {
+        const { _participant } = this.props;
+        const { id } = _participant;
+        const styles = this._getStyles();
+
+        return (
+            <div
+                className = 'avatar-container'
+                style = { styles.avatarContainer }>
+                <Avatar
+                    className = 'userAvatar'
+                    participantId = { id } />
+            </div>
+        );
+    }
+
+    /**
+     * Renders the local participant's thumbnail.
+     *
+     * @returns {ReactElement}
+     */
+    _renderLocalParticipant() {
+        const {
+            _defaultLocalDisplayName,
+            _disableProfile,
+            _participant,
+            _videoTrack
+        } = this.props;
+        const { id } = _participant || {};
+        const { audioLevel = 0 } = this.state;
+
+
+        return (
+            <>
+                <div className = 'videocontainer__background' />
+                <span id = 'localVideoWrapper'>
+                    <VideoTrack
+                        id = 'localVideo_container'
+                        videoTrack = { _videoTrack } />
+                </span>
+                <div className = 'videocontainer__toolbar'>
+                    <StatusIndicators participantID = { id } />
+                </div>
+                <div className = 'videocontainer__toptoolbar'>
+                    { this._renderTopToolbar() }
+                </div>
+                <div className = 'videocontainer__hoverOverlay' />
+                <div className = 'displayNameContainer'>
+                    <DisplayName
+                        allowEditing = { !_disableProfile }
+                        displayNameSuffix = { _defaultLocalDisplayName }
+                        elementID = 'localDisplayName'
+                        participantID = { id } />
+                </div>
+                { this._renderAvatar() }
+                <span className = 'audioindicator-container'>
+                    <AudioLevelIndicator audioLevel = { audioLevel } />
+                </span>
+            </>
+        );
+    }
+
+
+    /**
+     * Renders a remote participant's 'thumbnail.
+     *
+     * @returns {ReactElement}
+     */
+    _renderRemoteParticipant() {
+        const {
+            _audioTrack,
+            _participant,
+            _startSilent
+        } = this.props;
+        const { id } = _participant;
+        const { audioLevel = 0, volume = 1 } = this.state;
```

Comment:
About the `audioLevel` the situation is exactly the same as in the method for the local participant. See my previous comment.

About the `volume`:
We don't want to use `undefined` value because the volume of the element needs to be in sync with the slider. Of course we can modify the slider to handle `undefined` as `1` but I think with the default value it is a little bit clearer what is going on in the code.
Also when the audio element is mounted, `_onAudioElementReferenceChanged` will be called with the initial `volume` value which I guess could be `undefined`.

Reply:
According to https://html.spec.whatwg.org/multipage/media.html#dom-media-volume it should return a number. Can we defer handling undefined until we see it happen? I'm generally not a fan of being so defensive against potentially impossible cases.

Reply:
Agree! It should be fixed now!